modules/fedimint-ln-client/src/pay.rs

```diff
@@ -0,0 +1,536 @@
+            .send()
+            .await
+            .map(|res| {
+                info!("Payment response: {:?}", res);
+                res.status().is_success()
+            })
+            .expect("Error getting return value")
+    }
+
+    async fn transition_outgoing_contract_execution(
+        old_state: LightningPayStateMachine,
+        is_success: bool,
+        contract_id: ContractId,
+    ) -> LightningPayStateMachine {
+        // TODO: Retry contacting gateway depending on error?
+        if is_success {
+            info!("transition_outgoing_contract_execution to success");
+            LightningPayStateMachine {
+                common: old_state.common,
+                state: LightningPayStates::Success,
+            }
+        } else {
+            info!("transition_outgoing_contract_execution to refundable");
+            LightningPayStateMachine {
+                common: old_state.common,
+                state: LightningPayStates::Refundable(LightningPayRefundable { contract_id }),
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
+pub struct LightningPayRefundable {
+    contract_id: ContractId,
+}
+
+impl LightningPayRefundable {
+    fn transitions(
+        &self,
+        _common: &LightningPayCommon,
+        context: LightningClientContext,
+        global_context: DynGlobalClientContext,
+    ) -> Vec<StateTransition<LightningPayStateMachine>> {
+        let contract_id = self.contract_id;
+        vec![StateTransition::new(
+            Self::is_outgoing_contract_refundable(self.contract_id, global_context.clone()),
+            move |dbtx, refundable, old_state| {
+                Box::pin(Self::transition_refund(
+                    old_state,
+                    refundable,
+                    contract_id,
+                    dbtx,
+                    context.clone(),
+                    global_context.clone(),
+                ))
+            },
+        )]
+    }
+
+    async fn transition_refund(
+        old_state: LightningPayStateMachine,
+        refundable: bool,
+        contract_id: ContractId,
+        dbtx: &mut ClientSMDatabaseTransaction<'_, '_>,
+        context: LightningClientContext,
+        global_context: DynGlobalClientContext,
+    ) -> LightningPayStateMachine {
+        if refundable {
+            match Self::try_refund_outgoing_contract(
+                old_state.clone(),
+                contract_id,
+                dbtx,
+                context,
+                global_context,
+            )
+            .await
+            {
+                Ok(txid) => LightningPayStateMachine {
+                    common: old_state.common,
+                    state: LightningPayStates::Refund(LightningPayRefund { refund_txid: txid }),
+                },
+                Err(_) => LightningPayStateMachine {
+                    common: old_state.common,
+                    state: LightningPayStates::Failure,
+                },
```

Comment:
I think we'll need to retry here if, for example, the refund failed because the federation is offline

Reply:
An offline federation should not make the transaction submission state machine fail (that would be a bug). We will keep submitting the refund tx until the federation either tells us that it's invalid or that it was accepted.

Reply:
Yeah the `Failure` case here is when the refund tx is actually rejected by the federation, so this requires the federation to be online.